api/validation.go

```diff
@@ -60,10 +60,20 @@ func (api *API) validateAndHydrateDimensions(v dataset.Version, dims []model.Dim
 	}
 
 	if v.IsBasedOn.Type == cantabularMultivariateTable {
-		resp, err := api.getCantabularDimensions(ctx, dims, pType)
+
+		if postDimension {
+			hydrated, err := api.HydrateMultivariateDimensions(dims, pType)
+			if err != nil {
+				return nil, err
+			}
+			return hydrated, nil
+		}
+
+		resp, err := api.getCantabularDimensions(ctx, dims, pType, postDimension)
```

Comment:
This is bizzarre. You're passing the postDimension bool here but if postDimension == true it will never reach here. My instinct is that passing a 'PostDimension' bool around these validation functions isn't the correct approach. It's certainly not scalable (we can't just add another bool for each endpoint that needs something different done). Is there no way to rearrange it so the extra logic for PostDimension is handled in the PostDimensionHandler?

Reply:
Seems like maybe after retrieving the response from ValidateAndHydrateDimensions you could then put it through the logic for replacing with default categorizations